core/src/main/scala/akka/kafka/internal/SubSourceLogic.scala

```diff
@@ -57,16 +59,24 @@ private[kafka] abstract class SubSourceLogic[K, V, Msg](
     }
   }
 
-  val partitionAssignedCB = getAsyncCallback[Iterable[TopicPartition]] { tps =>
-    pendingPartitions ++= tps.filter(!partitionsInStartup.contains(_))
-    pump()
+  def partitionAssignedCB(kafkaConsumer: KafkaConsumer[_, _], tps: Iterable[TopicPartition]) = {
```

Comment:
As I understand it, those async callbacks must be removed. KafkaConsumerActor calls `poll`, `onPartitionsAssigned` is called, `seek` must be called before returning from `onPartitionsAssigned`. That means that all that must be in the KafkaConsumerActor without any asyncCallback.

Reply:
I actually think it's okay to put the call to `pump()` inside the async callback. If the `KafkaConsumer` has been assigned partitions, but `pump()` hasn't been called yet, it'll just mean that no downstream `Source`s will send a request for messages to the `KafkaConsumerActor` yet which is fine, it'll just pause the partition until it does receive a request.